File: Models/BlogModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace MyWeb.Models
{
    public class BlogPost
    {
        public int Id { get; set; }
        [Required(ErrorMessage = "標題為必填")]
        [StringLength(200)]
        public string Title { get; set; }
        public int CategoryId { get; set; }
        public BlogCategory Category { get; set; }
        public string Summary { get; set; }
        public string Content { get; set; }
        public string ImagePath { get; set; }
        public string Author { get; set; }
        public int ViewCount { get; set; }
        public bool IsPublished { get; set; }
        public DateTime? PublishedAt { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public string Tags { get; set; }
    }

    public class BlogCategory
    {
        public int Id { get; set; }
        [Required]
        [StringLength(100)]
        public string Name { get; set; }
        public int SortOrder { get; set; }
        public bool IsActive { get; set; }
    }

    public class BlogListViewModel
    {
        public IEnumerable<BlogPost> Posts { get; set; }
        public IEnumerable<BlogCategory> Categories { get; set; }
        public int? CurrentCategoryId { get; set; }
        public string SearchKeyword { get; set; }
        public int TotalCount { get; set; }
        public int PageIndex { get; set; }
        public int PageSize { get; set; }
    }
}

File: Models/BlogModel.vb
Imports System.ComponentModel.DataAnnotations

Namespace My.Models

    Public Class BlogPost
        Public Property Id As Integer
        <Required(ErrorMessage:="標題為必填")>
        <StringLength(200)>
        Public Property Title As String
        Public Property CategoryId As Integer
        Public Property Category As BlogCategory
        Public Property Summary As String
        Public Property Content As String
        Public Property ImagePath As String
        Public Property Author As String
        Public Property ViewCount As Integer
        Public Property IsPublished As Boolean
        Public Property PublishedAt As Date?
        Public Property CreatedAt As Date
        Public Property UpdatedAt As Date
        Public Property Tags As String
    End Class

    Public Class BlogCategory
        Public Property Id As Integer
        <Required>
        <StringLength(100)>
        Public Property Name As String
        Public Property SortOrder As Integer
        Public Property IsActive As Boolean
    End Class

    Public Class BlogListViewModel
        Public Property Posts As IEnumerable(Of BlogPost)
        Public Property Categories As IEnumerable(Of BlogCategory)
        Public Property CurrentCategoryId As Integer?
        Public Property SearchKeyword As String
        Public Property TotalCount As Integer
        Public Property PageIndex As Integer
        Public Property PageSize As Integer
    End Class

End Namespace
